File: backend/Multibanca.Domain.Models/Multibanca/BBVA/Escrituracion/firmar_rep_legal.cs
namespace Multibanca.Domain.Models.Multibanca.BBVA.Escrituracion;

public class firmar_rep_legal
{
    public long     id { get; set; }
    public long     id_expediente { get; set; }
    public string?  id_actividad { get; set; }

    // Campos de la actividad
    public string?  concepto_firma { get; set; }
    public string?  tipologia { get; set; }
    public string?  casuistica { get; set; }
    public string?  observaciones { get; set; }

    // Auditoría
    public bool      is_active { get; set; }
    public bool      row_status { get; set; }
    public int       created_by { get; set; }
    public DateTime  created_date { get; set; }
    public int?      modified_by { get; set; }
    public DateTime? modified_date { get; set; }
}

File: frontend/src/features/actividades/firmar_rep_legal/models/firmar_rep_legal.ts
import type { Auditoria } from '@/models/Auditoria';

export interface FirmarRepLegal extends Auditoria {
  id: number;
  id_expediente: number;
  id_actividad: string;
  concepto_firma: string | null;
  tipologia: string | null;
  casuistica: string | null;
  observaciones: string | null;
}

export const EMPTY_FIRMAR_REP_LEGAL = (
  id_expediente: number,
): FirmarRepLegal => ({
  id: 0,
  id_expediente,
  id_actividad: 'BBVA_ESCRITURACION_FIRMAR_REP_LEGAL',
  concepto_firma: null,
  tipologia: null,
  casuistica: null,
  observaciones: null,
  is_active: true,
  row_status: true,
  created_by: 0,
  created_date: '',
  modified_by: null,
  modified_date: null,
});
